### PR TITLE
test(server): pin URL-over-body binding precedence as a tested contract

### PR DESCRIPTION
**Pull request by gaborage**
## What

The request binder overlays path param, query, and header on top of the JSON body, so a present URL value beats a conflicting body value for a field carrying both a `json` tag and a URL tag. Nothing tested it, and it is not the engine's default — echo binds the body last, so go-bricks' post-body overlay is the whole mechanism. ADR-001 recorded the ordering but stated it without the guard that the overlay only fires when the value is present.

## Impact

None at runtime — the only `server/handler.go` edits are doc comments. One consumer-facing correction: a dual-tagged identifier is **not** a mass-assignment control on its own. An empty path segment still matches its route, so `POST /cards//status` against `/cards/:cardId/status` binds `cardId` as `""` and the body's value survives. Give a security-sensitive identifier only its URL tag.

## Verification

The tests bite: disabling the overlay turns the path, query, and header cases red with the body's values and leaves the guard cases green. The empty-segment case was confirmed against the real router, not reasoned about.

## Summary by CodeRabbit

- **Behavior**
  - Clarified request-binding precedence when fields are supplied by multiple sources.
  - JSON body values are applied first, then present path, query, and header values override conflicting values.
  - Empty path segments and absent or empty query values preserve body data; present-but-empty headers can clear slice fields.

- **Documentation**
  - Added guidance covering binding behavior across HTTP methods, including security considerations for URL-only identifiers.

- **Tests**
  - Added coverage for source precedence and boundary cases involving empty or absent URL and header inputs.

### CODE REVIEW

File: server/handler_test.go [hunk 1]
```diff
@@ -196,6 +196,155 @@ func TestRequestBinderAdvancedBinding(t *testing.T) {
...
+
+	// No `limit` query param; body supplies limit=999. Body value survives.
+	req := httptest.NewRequestWithContext(context.Background(), http.MethodPost, "/x", strings.NewReader(`{"limit":999}`))
+	req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+
+	require.NoError(t, h(c))
+	assert.Equal(t, uint16(999), seen.Limit, "absent query param must not clobber the body value")
+}
```
[the remark this answers is not in the document]
Comment (gaborage) [reply server/handler_test.go:317]: Fixed in 2db4302.

You were right, and probing the cases turned up a documentation error rather than just a coverage gap. The two guards differ in shape:

- `bindQueryValue` guards on the **value**, so both an absent key and a present-but-empty `?limit=` leave the body value in place.
- `bindHeaderValue` guards on `Header.Values` being **empty**, which only covers an *absent* header. A present-but-empty header is `[""]` — length 1 — so it reaches `bindHeaderStringSlice`, parses no entries, and still calls `Set`, clearing a `[]string` field instead of falling back to the body.

The wiki bullet had generalized query and header into one sentence and was wrong about that last case; it now states each guard separately.

Three tests added, matching the cases you named:

- `TestRequestBinderPrecedenceEmptyQueryParamKeepsBody` — `?limit=` present but empty
- `TestRequestBinderPrecedenceAbsentHeaderKeepsBody` — no `X-Items`
- `TestRequestBinderPrecedenceEmptyHeaderClearsSliceField` — empty `X-Items`

The last one pins the clearing behavior rather than endorsing it; whether the binder should fall back to the body there is a behavior change and belongs in its own PR.

The branch was also rebased onto main (`fa5f791`), replacing the earlier merge commit.
